third_party/ijar/zip_main.cc

```diff
@@ -152,11 +152,14 @@ int extract(char *zipfile, char *exdir, char **files, bool verbose,
     return -1;
   }
 
-  char output_root[PATH_MAX];
+  char output_root[PATH_MAX + 1];
   if (exdir != NULL) {
-    concat_path(output_root, PATH_MAX, cwd.c_str(), exdir);
+    concat_path(output_root, sizeof(output_root), cwd.c_str(), exdir);
```

Comment:
Shouldn't we also check if cwd.length()+strlen(exdir) >= sizeof(output_root)?

Reply:
Or alternatively always properly truncate, which is then `strncpy(..., PATH_MAX-1)`

Reply:
Using PATH_MAX-1 is probably not right, since it is unspecified if PATH_MAX has room for the nul or not on all systems.  Increasing the buffer by 1 always leaves room.

Reply:
@aiuto then you're ok that the value is truncated when `exdir` is set and error returned when `cwd` is too long? but not error returned when `exdir+cwd` is too long?

Reply:
I agree `concat_path` will not behave correctly if the paths are too long, but my goal was to fix possible memory corruption rather than that.

I will make `concat_path` return success boolean.

Reply:
Ok, since this is an improvement with respect to the previous code I'll approve and import it. Thanks @benjaminp.

Reply:
Yes. I was OK with the truncation because fixing that in a useful way (reporting the error and having bazel act on it) is a much larger PR.